Add aggegration toggle for clusterRoles

sandert-k8s opened the pull request:
## Explanation
Basically, this is the exact same PR as #11688 , but I wasn't allowed to push the necessary updates to this fork. So I recreated it. Thanks to @allanp35 for the initial commits.

We wanted to have the possibility to unable aggregate cluster role to admin, edit and view cluster roles. This feature is essential for us because we used those cluster roles for our users in namespaces but we want to block the possibility for our users to create kyverno policies for example.

## Related issue

## Milestone of this PR

## Documentation (required for features)

My PR contains new or altered behavior to Kyverno. 
- [ ] I have sent the draft PR to add or update [the documentation](https://github.com/kyverno/website) and the link is:
  <!-- Uncomment to link to the PR -->
  <!-- https://github.com/kyverno/website/pull/123 -->

## What type of PR is this

<!--

> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading white spaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
-->
/kind feature
<!--
-->

## Proposed Changes

We wanted to have the possibility to unable aggregate cluster role to admin, edit and view cluster roles. This feature is essential for us because we used those cluster roles for our users in namespaces but we want to block the possibility for our users to create kyverno policies for example.

### Proof Manifests

Proof Manifests
This PR is used to enable or not the presence of aggregate label, as in this Cluster Role :

Kubernetes resource
```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  labels:
    app.kubernetes.io/component: rbac
    app.kubernetes.io/instance: kyverno
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/part-of: kyverno
    app.kubernetes.io/version: 3.1.4
    helm.sh/chart: kyverno-3.1.4
    rbac.authorization.k8s.io/aggregate-to-admin: 'true'
  name: 'kyverno:rbac:admin:reports'
rules:
  - apiGroups:
      - kyverno.io
    resources:
      - admissionreports
      - clusteradmissionreports
      - backgroundscanreports
      - clusterbackgroundscanreports
    verbs:
      - create
      - delete
      - get
      - list
      - patch
      - update
      - watch
```
This PR doesn't delete d-the Cluster Role, put allow the ability to unable the aggregation to admin, edit or view role.

After this PR, ClusterRole can now be created as follows :

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  labels:
    app.kubernetes.io/component: rbac
    app.kubernetes.io/instance: kyverno
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/part-of: kyverno
    app.kubernetes.io/version: 3.1.4
    helm.sh/chart: kyverno-3.1.4
  name: 'kyverno:rbac:admin:reports'
rules:
  - apiGroups:
      - kyverno.io
    resources:
      - admissionreports
      - clusteradmissionreports
      - backgroundscanreports
      - clusterbackgroundscanreports
    verbs:
      - create
      - delete
      - get
      - list
      - patch
      - update
      - watch
```

## Checklist

<!--
Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
-->

- [x] I have read the [contributing guidelines](https://github.com/kyverno/kyverno/blob/main/CONTRIBUTING.md).
- [x] I have read the [PR documentation guide](https://github.com/kyverno/kyverno/blob/main/.github/pr_documentation.md) and followed the process including adding proof manifests to this PR.
- [ ] This is a bug fix and I have added unit tests that prove my fix is effective.
- [ ] This is a feature and I have added CLI tests that are applicable.
- [ ] My PR needs to be cherry picked to a specific release branch which is <replace>.
- [ ] My PR contains new or altered behavior to Kyverno and
  - [ ] CLI support should be added and my PR doesn't contain that functionality.

## Further Comments

sandert-k8s commented on config/crds/kyverno/kyverno.io_globalcontextentries.yaml:
```diff
@@ -19,6 +19,9 @@ spec:
   scope: Cluster
   versions:
   - additionalPrinterColumns:
+    - jsonPath: .status.conditions[?(@.type == "Ready")].status
```
This file is automatically editted by the `make codegen-helm-all` command. I'm not 100% sure it should do this; can a maintainer confirm?

realshuting commented:
Hi @sandert-k8s  - can you merge the latest main branch and run the code-gen again?

sandert-k8s commented:
Done @realshuting !

eddycharly commented:
@sandert-k8s can we do something more granular like:
```yaml
rbac:
  userFacingRoles:
    admin: true
    edit: true
    view: true
```

sandert-k8s commented:
@eddycharly Sounds good indeed, fixed that. The `edit` role isn't defined in the rbac section, so I didn't added it.

eddycharly commented:
@sandert-k8s do we need to generate the cluster roles at all when disabled ?

sandert-k8s commented:
@eddycharly I only try to disable the aggregation. To completely disable the rbac section would be another story/PR I think. 

But, the labels should always be added to the role, which weren't added if the value equalled false. I refactored it so it does now. 
I also changed the the value back to `aggregate`, since i don't want to disable the user-facing role, I only intend to disable the aggregation.

eddycharly commented:
> @eddycharly I only try to disable the aggregation. To completely disable the rbac section would be another story/PR I think.
> 
> But, the labels should always be added to the role, which weren't added if the value equalled false. I refactored it so it does now. I also changed the the value back to `aggregate`, since i don't want to disable the user-facing role, I only intend to disable the aggregation.

I mean those roles are dedicated to aggregation, what would be a good reason to keep them if they are not aggregated in the end ?

sandert-k8s commented:
> I mean those roles are dedicated to aggregation, what would be a good reason to keep them if they are not aggregated in the end ?

I could still assume that you manually create a rolebinding for it, and than it's convenient that there is already a role available to use.

eddycharly commented:
@sandert-k8s tests are failing, can you look into it ?

sandert-k8s commented:
@eddycharly Can you rerun the tests?

sandert-k8s commented:
@eddycharly I don't believe the failing chainsaw tests are caused by this change.
